packages/@react-spectrum/textfield/src/TextArea.tsx

```diff
@@ -42,17 +46,17 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
     if (inputRef.current) {
       // if no value or defaultValue is passed
       // no need to call onHeightChange
-      if (!inputRef.current.value) {
+      if (!inputValue || inputValue.length === 0) {
```

Comment:
I think this isn't quite right. If you go to the controlled story, and click the "Set Text" button and then select all and delete, the text area doesn't resize back down.